### DISCUSSION TITLE
Where can I find the v1 documentation?

### DISCUSSION THREAD

**Discussion by mrdulin** [reduxjs/redux-toolkit#3998] [Q&A] [answered]
It seems RTK version 2 is incompatible with TS 4.3.2, see https://stackoverflow.com/questions/77684107/got-error-ts1005-expected-when-try-to-run-a-type-check-command-using-tsc/77684170

Downgrading the RTK to v1.9.7 solves the problem. So I am looking for the RTK v1.9.7 documentation.

**Comment by markerikson** [comment, marked as answer]
Yes, we now require TS 4.7+ for RTK 2.0.

We don't maintain fully versioned docs. You'd need to look at the contents of the `docs` folder in the repo.

That said, there aren't that many differences API-wise between v1 and v2.  See the migration guide at https://redux-toolkit.js.org/usage/migrating-to-modern-redux for an idea of what's changed.

A couple questions:

- is there something specific you're wanting to know about, API-wise?
- any reason why you can't upgrade to TS 4.7?

**Reply by mrdulin** [reply]
1. I am worried there are some breaking changes
2. I am working on an old project, upgrade TS will cause other issues. Most of modules of the project use `redux` by now, I am going to use RTK in some modules

BTW, I didn't find RTK requirements for TS version compatibility in the documentation

**Reply by markerikson** [reply]
That's changed over time. RTK 1.9 requires at least TS 4.2+.

Honestly, I would go ahead and upgrade to at _least_ TS 4.7 in your project right now, if not TS 5+.

**Reply by mrdulin** [reply]
The project is too big and there is not enough time at the moment. Too much historical debt.

Maybe add more legacy type definition files in the `typeVersions` config of `package.json` for RTK v2?

**Reply by markerikson** [reply]
No, we're not going to try to do that. The point of RTK 2.0 was to drop a lot of legacy backwards compatibility pieces.

**Reply by phryneas** [reply]
To be honest, every minor of TypeScript you skip, you add more historical debt and make it harder to upgrade. TypeScript is one of those things you should always upgrade to the latest version when it's released, and it's usually very painless to jump just one version (most of the time no work, sometimes half an hour). Upgrading it for 5-6 versions at once can be more painful, but that's where you got yourself by not updating.

**Reply by mrdulin** [reply]
Thanks for your advice. I joined the project halfway through.

**Comment by kno-raziel** [comment]
Recommendation for your next major release, you could have done something similar to what React Router team did with the old version, the offer the old documentation this way `https://v5.reactrouter.com/` , you could have done `https://v1.redux-toolkit.js.org `. 

I was trying to type a listenerMiddleware using the documentation found in [https://redux-toolkit.js.org/api/createListenerMiddleware#startlistening](https://redux-toolkit.js.org/api/createListenerMiddleware#startlistening) just to realize I was looking at the wrong version.       

Gladly I found the answer in StackOverflow ✌️ [https://stackoverflow.com/a/74814320](https://stackoverflow.com/a/74814320) if not I probably should have spend time looking for it the docs folder in repo in the old versions...

**Reply by EskiMojo14** [reply]
Out of curiosity, what's stopping you from updating to v2?

**Reply by markerikson** [reply]
It's certainly _possible_, but it's also considerable extra effort to set that up. Given the fairly straightforward migration steps to use v2, and the amount of effort it would take, it's not worth it for us to set that up.

**Reply by kno-raziel** [reply]
> Out of curiosity, what's stopping you from updating to v2?

Isa legacy project with some tech debt, right now upgrade libraries is not like a priority 🫠
